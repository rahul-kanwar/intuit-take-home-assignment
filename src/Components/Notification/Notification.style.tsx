import styled from 'styled-components';

export const NotificationContainer = styled.div`
  position: fixed;
  top: 10px;
  opacity: 0.8;
  width: 320px;
  background-color: #4caf50;
  color: white;
  padding: 15px;
  border-radius: 5px;
  box-shadow: 0 0 10px rgba(0, 0, 0, 0.1);
  z-index: 1000;
`;
