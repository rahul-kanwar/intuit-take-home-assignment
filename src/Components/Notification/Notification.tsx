import React, { useEffect } from 'react';

import { NotificationContainer } from './Notification.style';

interface NotificationProps {
  message: string;
  onClose: () => void;
}

const Notification: React.FC<NotificationProps> = ({ message, onClose }) => {
  useEffect(() => {
    const timer = setTimeout(() => {
      onClose();
    }, 3000);

    return () => clearTimeout(timer);
  }, [onClose]);

  return <NotificationContainer>{message}</NotificationContainer>;
};

export default Notification;
