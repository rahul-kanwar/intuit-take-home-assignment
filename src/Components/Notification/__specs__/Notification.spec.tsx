import React from 'react';
import { render, waitFor, screen } from '@testing-library/react';
import 'jest-styled-components';
import { act } from 'react-dom/test-utils'; 
import Notification from '../Notification';

describe('Notification Component', () => {
  it('renders correctly', () => {
    const { asFragment } = render(<Notification message="Test Notification" onClose={() => {}} />);
    expect(asFragment()).toMatchSnapshot();
  });

  it('displays the correct message', () => {
    render(<Notification message="Test Notification" onClose={() => {}} />);
    expect(screen.getByText('Test Notification')).toBeInTheDocument();
  });

  it('auto-closes after 3 seconds', async () => {
    const mockOnClose = jest.fn();
    render(<Notification message="Test Notification" onClose={mockOnClose} />);

    await waitFor(() => {
      expect(mockOnClose).toHaveBeenCalled();
    }, { timeout: 3000 });
  });
});