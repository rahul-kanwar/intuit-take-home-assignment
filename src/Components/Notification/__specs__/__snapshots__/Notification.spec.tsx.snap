// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Notification Component renders correctly 1`] = `
<DocumentFragment>
  .c0 {
  position: fixed;
  top: 10px;
  opacity: 0.8;
  width: 320px;
  background-color: #4caf50;
  color: white;
  padding: 15px;
  border-radius: 5px;
  box-shadow: 0 0 10px rgba(0, 0, 0, 0.1);
  z-index: 1000;
}

<div
    class="c0"
  >
    Test Notification
  </div>
</DocumentFragment>
`;
